src/ia32/lithium-codegen-ia32.cc

```diff
@@ -6532,6 +6476,92 @@ void LCodeGen::DoUnarySIMDOperation(LUnarySIMDOperation* instr) {
 void LCodeGen::DoBinarySIMDOperation(LBinarySIMDOperation* instr) {
   uint8_t imm8 = 0;  // for with operation
   switch (instr->op()) {
+    case kFloat32x4ExtractLane: {
+      DCHECK(instr->hydrogen()->left()->representation().IsFloat32x4());
+      DCHECK(instr->hydrogen()->right()->representation().IsInteger32());
+      XMMRegister left_reg = ToFloat32x4Register(instr->left());
+      uint32_t right = ToInteger32(LConstantOperand::cast(instr->right()));
```

Comment:
You should check whether the right operand is constant or not. Deoptimize if it is not a constant.